iron-location.html

```diff
@@ -231,9 +231,14 @@
       if (!href) {
         return;
       }
+      event.preventDefault();
+      // If the navigation is to the current page we shouldn't add a history
+      // entry or fire a change event.
+      if (href === window.location.href) {
+        return;
+      }
       window.history.pushState({}, '', href);
       this.fire('location-changed', {}, {node: window});
-      event.preventDefault();
```

Comment:
It's just now moved above so that we can short circuit the pushing to history and notifying.